Sources/SwiftFusion/Core/Vector.swift

```diff
@@ -1,15 +1,21 @@
 import Foundation
 import TensorFlow
 
+protocol DifferentiableMutableCollection: Differentiable, MutableCollection {
+  @differentiable
+  var count: Int { get }
```

Comment:
This should not be `@differentiable` because it's not a `Differentiable` type.

If the compiler is complaining about `x.count` not being differentiable, you can do `withoutDerivative(at: x).count`.

Reply:
However, I think that all the methods using `.count` (`.flatTensor`) are ones whose implementations are not automatically differentiable and therefore the real solution is to make sure that the custom derivative implementation is registered correctly.

Reply:
> If the compiler is complaining about `x.count` not being differentiable, you can do `withoutDerivative(at: x).count`.

Alternatively, you can mark `var count` as `@noDerivative`. That achieves the same affect as `withoutDerivative(at: x).count` without complicating use sites, you can continue to use `x.count`.